docs/use_cases/retrieval_augmented_generation_eval_qdrant_ragas.md

```diff
@@ -0,0 +1,624 @@
+use later during the evaluation step.
+
+Though each of the techniques above - T5, OpenAI, and RAGAS - confers certain advantages and disadvantages, all of them let you build a good evaluation dataset including reasonably good questions and ground-truths to evaluate your RAG system. Which you decide to use depends on the specifics of your use case.
+
+### Tuning with answers
+
+In each tuning cycle (i.e., adjusting model parameters, training the model, evaluating the model), your evaluation dataset should be tuned using RAG pipeline-generated `answer`s. This requires developing a subroutine that facilitates the construction of an evaluation dataset in the expected RAGAS format (`question`, `answer`, `contexts`, `ground_truths` - as indicated in the above code snippet) executing the provided questions through your RAG system.
+
+Such a subroutine might look like this:
+
+```python
+## Prepare the evaluation dataset to evaluate our RAG system
+from datasets import Dataset
+
+# RAGAS Expect ['question', 'answer', 'contexts', 'ground_truths'] format
+'''
+{
+    "question": ['What is quantization?', ...], 
+    "answer": [], ## answer
+    "contexts": [], ## context
+    "ground_truths": [] ## answer expected
+}
+'''
+# loading the eval dataset from HF
+qdrant_qa = load_dataset("atitaarora/qdrant_doc_qna", split="train")
+
+questions = []
+answers = []
+contexts = []
+ground_truths = []
+
+for query in qdrant_qa:
+    question = query["question"]
+    questions.append(question)
+    ground_truths.append([query["answer"]])
+    
+    rag_response = query_with_context(question)
+    answers.append(rag_response)
+    search_result_context = client.query(collection_name=COLLECTION_NAME, query_text=question,limit=4)
+    context = ["document:"+r.document+",source:"+r.metadata['source'] for r in search_result_context]
+    contexts.append(context)    
+
+rag_response_data = {
+    "question": questions,
+    "answer": answers,
+    "contexts": contexts,
+    "ground_truths": ground_truths
+}
+
+## This dataset 'rag_response_dataset', is created to evaluate the RAG system using the RAGAS format.
+## Note: this dataset is specifically for evaluation purposes. Therefore, it needs to be recreated every time changes are made to the RAG config.
+rag_response_dataset = Dataset.from_dict(rag_response_data)
+# The dataset is then exported as a CSV file, with a filename that includes details of the experiment for easy identification, such as the chunk size used in this case -  rag_response_dataset.to_csv('rag_chunk_512.csv')
+```
+
+The subroutine above uses an abstraction of the RAG pipeline method `query_with_context()`. The `rag_response_dataset` uses an older format in which `ground_truths` is a list of strings, while the latest format uses `ground_truth` as a single string value (representing the correct answer to a question). 
```

Comment:
@atarora 
Where will the "latest format" be used?

Reply:
Its just the field rename for the format , the older `ground_truths` expected list of strings and the newer `ground_truth` just expects the same content in a plain string format for the ease of use.
I mentioned it to ensure the reference was backward compatible.

Reply:
Actually , glad you asked this : I modified the example https://github.com/qdrant/qdrant-rag-eval/blob/master/workshop-rag-eval-qdrant-ragas/notebook/naive_rag_eval_qdrant_ragas.ipynb to use latest format.